pkg/controller/servicebindingrequest/binder_test.go

```diff
@@ -480,3 +483,61 @@ func Test_extraFieldsModifier(t *testing.T) {
 	})
 
 }
+
+func TestBindAndUnbind(t *testing.T) {
+
+	t.Run("bind and unbind", func(t *testing.T) {
+		ns := "binder"
+		name := "service-binding-request"
+		f := mocks.NewFake(t, ns)
+		sbr := f.AddMockedServiceBindingRequest(name, nil, "backingServiceResourceRef", "applicationResourceRef", deploymentsGVR, nil)
+		f.AddMockedUnstructuredDeployment("applicationResourceRef", nil)
+
+		binder := newBinder(
+			context.TODO(),
+			f.FakeDynClient(),
+			sbr,
+			[]string{},
+			testutils.BuildTestRESTMapper(),
+		)
+		require.NotNil(t, binder)
+
+		_, err := binder.search()
```

Comment:
`binder.search()` is invoked in `binder.bind()`, so not sure if we really needed here?

Reply:
Add a comment for this. We need to ensure there is no error in search().